histomicstk/utils/sample_pixels.py

```diff
@@ -0,0 +1,113 @@
+import large_image
+import numpy as np
+import scipy
+
+from .simple_mask import simple_mask
+
+
+def sample_pixels(slide_path, magnification, sample_percent,
+                  mapping_mag=1.25, min_coverage=0.1):
+    """Generates a sampling of pixels from a whole-slide image.
+
+    Useful for generating statistics or Reinhard color-normalization or
+    adaptive deconvolution. Uses mixture modeling approach to focus
+    sampling in tissue regions.
+
+    Parameters
+    ----------
+    slide_path : str
+        path and filename of slide.
+    magnification : double
+        Desired magnification for sampling (defaults to native scan
+        magnification).
+    sample_percent : double
+        Percentage of pixels to sample. Must be in the range [0, 1].
+    mapping_mag: double, optional
+        low resolution magnification. Default value = 1.25.
+    min_coverage: double, optional
+        minimum sample_percent of tile covered by tissue to be included in sampling.
+        Ranges between [0,1). Default value = 0.1.
+
+    Returns
+    -------
+    Pixels : array_like
+        A 3xN matrix of RGB pixel values sampled from the whole-slide.
+
+    See Also
+    --------
+    histomicstk.preprocessing.color_normalization.reinhard,
+    histomicstk.preprocessing.color_deconvolution.SparseColorDeconvolution
+    """
+
+    ts = large_image.getTileSource(slide_path)
+
+    # get enitre whole-silde image at low resolution
+    scale_lres = {'magnification': mapping_mag}
+    im_lres, _ = ts.getRegion(
+        format=large_image.tilesource.TILE_FORMAT_NUMPY,
+        scale=scale_lres
+    )
+    im_lres = im_lres[:, :, :3]
+
+    # compute foreground mask of whole-slide image at low-res
+    im_fgnd_mask_lres = simple_mask(im_lres)
+
+    # generate sample pixels
+    sample_pixels = []
+
+    scale_hres = {'magnfication': magnification}
+
+    for tile in ts.tileIterator(
+            scale=scale_hres,
+            format=large_image.tilesource.TILE_FORMAT_NUMPY):
+
+        # get current region in base_pixels
+        rgn_hres = {'left': tile['gx'], 'top': tile['gy'],
+                    'right': tile['gx'] + tile['gwidth'],
+                    'bottom': tile['gy'] + tile['gheight'],
+                    'units': 'base_pixels'}
```

Comment:
Would be nice if i can pull this `rgn_hres` from `tile` instead of having to construct it like this.